proteus/mprans/RANS2P.py

```diff
@@ -1460,216 +1461,226 @@ def getResidual(self, u, r):
-                                      self.coefficients.wettedAreas,
-                                      self.coefficients.netForces_p,
-                                      self.coefficients.netForces_v,
-                                      self.coefficients.netMoments,
-                                      self.q['velocityError'],
-                                      self.velocityErrorNodal,
-                                      self.q[('force', 0)],
-                                      self.q[('force', 1)],
-                                      self.q[('force', 2)],
-                                      int(self.coefficients.use_ball_as_particle),
-                                      self.coefficients.ball_center,
-                                      self.coefficients.ball_radius,
-                                      self.coefficients.ball_velocity,
-                                      self.coefficients.ball_angular_velocity,
-                                      self.coefficients.ball_center_acceleration,
-                                      self.coefficients.ball_angular_acceleration,
-                                      self.coefficients.ball_density,
-                                      self.coefficients.particle_signed_distances,
-                                      self.coefficients.particle_signed_distance_normals,
-                                      self.coefficients.particle_velocities,
-                                      self.coefficients.particle_centroids,
-                                      self.coefficients.ebq_global_phi_s,
-                                      self.coefficients.ebq_global_grad_phi_s,
-                                      self.coefficients.ebq_particle_velocity_s,
-                                      self.coefficients.nParticles,
-                                      self.coefficients.particle_netForces,
-                                      self.coefficients.particle_netMoments,
-                                      self.coefficients.particle_surfaceArea,
-                                      int(self.mesh.nElements_owned),
-                                      self.coefficients.particle_nitsche,
-                                      self.coefficients.particle_epsFact,
-                                      self.coefficients.particle_alpha,
-                                      self.coefficients.particle_beta,
-                                      self.coefficients.particle_penalty_constant,
-                                      self.coefficients.phi_s,
-                                      self.coefficients.phisField,
-                                      self.coefficients.use_pseudo_penalty,
-                                      self.coefficients.useExact)
+
+        argsDict = cArgumentsDict.ArgumentsDict()
+        argsDict["NONCONSERVATIVE_FORM"] = self.coefficients.NONCONSERVATIVE_FORM
```

Comment:
NONCONSERVATIVE_FORM and MOMENTUM_SGE, etc. are real valued flags, but one of the tests sets them to integers. Maybe just wrap them in `float()~ so reasonable dynamic casts work (e.g. bool or int promotion to float will be fine in all cases).

Reply:
OK, looks like also `argsDict["shockCapturingDiffusion"] = int(self.shockCapturing.shockCapturingFactor)` should use float instead of int. That's another postive real number.